### PR TITLE
Add tests and document that `currentPath` of navigation can be an array of strings.

### LINKED ISSUE #2422: Navigation: Current page is not marked for nested sub-navigation
In the following example it is not possible to have both "Guidance" (the parent section) and "ONS sub-branding" (the page) marked as the current page at the same time:

Image

A solution might be to allow an array of strings to be provided for `currentPath` such that multiple navigation items can be marked as current. If a string is provided (as with the current implementation) then the macro would wrap that inside an array so that the macro implementation only needs to deal with arrays.

### PR DESCRIPTION
**Pull request by kruncher**
### What is the context of this PR?
Resolves #2422, #2424 and #2428

### How to review
- Review documentation.
- Verify that unit tests pass.

### CODE REVIEW

File: src/components/header/_macro-options.md [hunk 1]
```diff
@@ -44,7 +44,7 @@
 | ariaLabel    | string                | false    | The `aria-label` attribute added to the `<nav>` element. Defaults to “Main menu”.                                                          |
 | itemsList    | array`<Item>`         | true     | Settings for an array of [list items](#item) for each navigation link                                                                      |
 | siteBasePath | string                | false    | If the root path for your site is not `/` (i.e. if using the prototype kit), you can set this path to the home page to set the active page |
-| currentPath  | string                | true     | The path of the current active page                                                                                                        |
+| currentPath  | string or string[]    | true     | The path of the current active page. Multiple paths can be provided using an array to highlight nested navigation.                         |
```
Comment (jrbarnes9) [inline src/components/header/_macro-options.md:47]: For `params.variants` we document the type as: "array or string"

**Review by jrbarnes9 — APPROVED** [review]

**Review by jrbarnes9 — APPROVED** [review]

**Review by jrbarnes9 — APPROVED** [review]

**Comment by rmccar** [comment]
I think this ability to set multiple levels as the current path in navigation should also be added to the section navigation component and the vertical version example for that should show this working

File: src/components/navigation/_macro.njk [hunk 2]
```diff
@@ -79,14 +79,14 @@
...
                                 {% if item.sections %}
                                     {% for section in item.sections %}
                                         {% if section.sectionTitle %}<h3 class="ons-navigation__list-header">{{ section.sectionTitle }}</h3>{% endif %}
                                         <ul class="ons-navigation__list ons-navigation__list--child ons-list--dashed ons-u-ml-s ons-u-mt-xs">
                                             {% for child in section.children %}
-                                                <li class="ons-navigation__item ons-list__item {{ ' ons-navigation__item--active' if (child.url == params.navigation.subNavigation.currentPath) or (child.url != (params.navigation.subNavigation.sitsiteBasePatheBasePath | default('/')) and child.url in params.navigation.subNavigation.currentPath) }}">
+                                                <li class="ons-navigation__item ons-list__item {{ ' ons-navigation__item--active' if (child.url == params.navigation.currentPath) or (params.navigation.currentPath is not string and child.url in params.navigation.currentPath)}}">
```
Comment (rmccar) [inline src/components/navigation/_macro.njk:89]:
```suggestion
                                                <li class="ons-navigation__item ons-list__item {{ ' ons-navigation__item--active' if (child.url == params.navigation.currentPath) or (params.navigation.currentPath is not string and child.url in params.navigation.currentPath) }}">
```
Really small nit but this space shouldn't be taken out

**Comment by boxadesign** [comment]
> I think this ability to set multiple levels as the current path in navigation should also be added to the section navigation component and the vertical version example for that should show this working

We considered this but currently there is no requirement to do this on section nav

**Review by jrbarnes9 — APPROVED** [review]

**Review by boxadesign — APPROVED** [review]

**Pull request merged**